django/contrib/auth/__init__.py

```diff
@@ -199,12 +199,26 @@ def get_user(request):
             # Verify the session
             if hasattr(user, "get_session_auth_hash"):
                 session_hash = request.session.get(HASH_SESSION_KEY)
-                session_hash_verified = session_hash and constant_time_compare(
-                    session_hash, user.get_session_auth_hash()
+                session_auth_hash = user.get_session_auth_hash()
+                session_hash_verified = constant_time_compare(
+                    session_hash,
+                    session_auth_hash,
```

Comment:
Should we still be checking that `session_hash` is truthy here? For example:

```python
session_auth_hash = user.get_session_auth_hash()
session_hash_verified = session_hash  and constant_time_compare(
    session_hash,
    session_auth_hash,
)
```

Reply:
Good catch :dart: We shouldn't even call `get_session_auth_hash()` on non-hash sessions.